brave/brave-ios
# Fix #6334: add segment of NFT that allows users to add non-fungible custom assets.

## Summary of Changes
We introduce a segment on top of Add Custom Assets to distinguish fungible token and non-fungible token


This pull request fixes #6334

## Submitter Checklist:


## Test Plan:
1. open Brave and go to Brave wallet
2. set up a wallet or unlock your wallet if its locked 
3. click `Edit Visible Assets`
4. click `Add custom Asset`
5. observe there is a segment with two options: TOKEN and NFT
6. Check input entries under TOKEN. they should be 1. select network 2. token name 3. token address 4. token symbol 5. decimals of precision Advanced(1. icon url 2. coingecko id)
7. Check input entries under NFT. default it should be 1. select network 2. token name 3. token address 4. token symbol 5. token id
8. Under NFT, if you choose Ethereum or EVM network, input entries should be the same as the default one
9. Under NFT, if you choose Solana network, input entries should become 1. network 2. token name 3. token address 4. token symbol 
10. Check inputs will be cleared out if you switch your selected segment 
11. Check the `Add` button will be disabled if you missing required fields. For tokens: network, name, symbol, decimals, address(address has to be Eth address if Eth or EVM network has been selected) For NFT: network, name, symbol, tokenId(if ETH or EVM network has been selected), address(address has to be Eth address if Eth or EVM network has been selected)
12. Check Fungible or non-Fungible token can be successfully added buy checking if the token appears in visible assets list in Portfolio Screen in the correct network (token will be added as visible)

## Screenshots:


![Simulator Screen Shot - iPhone 14 Pro - 2022-11-09 at 10 48 14](https://user-images.githubusercontent.com/1187676/200876313-c16f0733-1813-42fe-98b2-3c140f3a3cdf.png) | ![Simulator Screen Shot - iPhone 14 Pro - 2022-11-09 at 10 48 18](https://user-images.githubusercontent.com/1187676/200876318-34c8d55b-0995-45e1-9c77-d1475ffc2f59.png) | ![Simulator Screen Shot - iPhone 14 Pro - 2022-11-09 at 10 48 22](https://user-images.githubusercontent.com/1187676/200876320-3837edb6-7413-407c-82cc-8bf192d72892.png)
--|--|--

## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
